### PR TITLE
Configure author terraform to work with DynamoDB.

### PR DESCRIPTION
**Pull request by samiwel**
Some changes were required to author's terraform scripts to support running author against a DynamoDB data store.

### How to test
`terraform init --upgrade`
`terraform plan`
`terraform apply`
Ensure you can create a questionnaire in author and launch a preview in runner.
Log into the AWS dev console and confirm that dynamo db tables exist for your environment.
Tables should contain the created questionnaire version.

### CODE REVIEW

File: author.tf [hunk 1]
```diff
@@ -384,30 +420,8 @@ module "author-schema-validator" {
...
-  database_password                = "${var.author_database_password}"
-  db_subnet_group_name             = "${module.author-vpc.database_subnet_group_name}"
-  database_identifier              = "${var.env}-authorrds"
-  rds_security_group_name          = "${var.env}-author-rds-access"
-  snapshot_identifier              = "pre-migrate-author"
-}
-
 module "author-survey-runner-dynamodb" {
-  source                                 = "github.com/ONSdigital/eq-terraform-dynamodb?ref=v2.2"
+  source                                 = "github.com/ONSdigital/eq-terraform-dynamodb?ref=973-deploy-author-running-against-dynamodb-to-aws"
```
Comment (jonnyshaw89) [inline author.tf]: It might be wise to use a different repo as we would need to create all of the runner tables just to run author

**Review by tgandrews — APPROVED** [review]
LGTM and works e2e

**Comment by samiwel** [comment]
@tgandrews there are enough changes here since you approved that it probably warrants another sanity check.

**Comment by tgandrews** [comment]
@samiwel @jonnyshaw89
Getting an error:
```
Error getting plugins: module root:
    module 'author-survey-runner': used_jti_claim_table_name is not a valid output for module author-survey-runner-dynamodb
    module 'author-survey-runner': questionnaire_state_table_arn is not a valid output for module author-survey-runner-dynamodb
    module 'author-survey-runner': eq_session_table_arn is not a valid output for module author-survey-runner-dynamodb
    module 'author-survey-runner': questionnaire_state_table_name is not a valid output for module author-survey-runner-dynamodb
    module 'author-survey-runner': submitted_responses_table_name is not a valid output for module author-survey-runner-dynamodb
    module 'author-survey-runner': eq_session_table_name is not a valid output for module author-survey-runner-dynamodb
    module 'author-survey-runner': submitted_responses_table_arn is not a valid output for module author-survey-runner-dynamodb
    module 'author-survey-runner': used_jti_claim_table_arn is not a valid output for module author-survey-runner-dynamodb
```

File: author.tf [hunk 2]
```diff
@@ -384,34 +423,12 @@ module "author-schema-validator" {
...
-  snapshot_identifier              = "pre-migrate-author"
-}
-
 module "author-survey-runner-dynamodb" {
-  source                                 = "github.com/ONSdigital/eq-terraform-dynamodb?ref=v2.2"
-  env                                    = "${var.env}-author"
-  aws_account_id                         = "${var.aws_account_id}"
-  aws_assume_role_arn                    = "${var.aws_assume_role_arn}"
-  slack_alert_sns_arn                    = "${module.eq-alerting.slack_alert_sns_arn}"
+  source              = "github.com/ONSdigital/eq-author-terraform-dynamodb?ref=973-deploy-author-running-against-dynamodb-to-aws"
```
Comment (jonnyshaw89) [inline author.tf]: This module now only deploy the author tables, so you need to keep the existing module for the runner tables

**Review by jonnyshaw89 — DISMISSED** [review]

**Comment by samiwel** [comment]
@tgandrews I think I forgot to push up a change to fix that yesterday. Want to give it another go?

File: author.tf [hunk 3]
```diff
@@ -310,19 +310,56 @@ module "author-api" {
...
+              "dynamodb:GetItem",
+              "dynamodb:DeleteItem"
+          ],
+          "Resource": "${module.author-dynamodb.author_questionnaires_table_arn}"
+      },
+      {
+          "Sid": "",
+          "Effect": "Allow",
+          "Action": [
+              "dynamodb:DescribeTable",
```
Comment (jonnyshaw89) [inline author.tf:358]: what is `dynamodb:DescribeTable` used for?

Comment (samiwel) [reply author.tf:358]: Dynamoose required this permission when defining the models.

**Review by tgandrews — APPROVED** [review]

**Comment by adammahmood** [comment]
Tested locally. LGTM 👍

**Review by adammahmood — APPROVED** [review]
Tested locally. LGTM 👍

**Review by rmccar — APPROVED** [review]
LGTM

**Comment by samiwel** [comment]
@jonnyshaw89 you still have a blocking review on this PR. Are you happy with the changes I've made? I have re-instated the RDS config so we can merge this safely before the dynamo branch is merged. Will remove the postgres DB in a separate PR.

File: developer_defaults.tf [hunk 4]
```diff
@@ -142,7 +142,7 @@ variable "author_registry" {
 
 variable "author_tag" {
   description = "The tag for the Author image to run"
-  default     = "latest"
+  default     = "797-replace-relational-database-with-json-document"
```
Comment (jonnyshaw89) [inline developer_defaults.tf]: Is this to be changed before merge?

File: author.tf [hunk 5]
```diff
@@ -321,8 +321,50 @@ module "author-api" {
...
+      {
+          "Sid": "",
+          "Effect": "Allow",
+          "Action": [
+              "dynamodb:Scan",
+              "dynamodb:DescribeTable",
+              "dynamodb:PutItem",
+              "dynamodb:UpdateItem",
+              "dynamodb:GetItem",
+              "dynamodb:DeleteItem"
```
Comment (jonnyshaw89) [inline author.tf:350]: If we can delete from `questionnaires_table` should it not also delete from `questionnaire_versions`?

File: developer_defaults.tf (hunk 4 above)
Comment (tgandrews) [inline developer_defaults.tf]:
```suggestion
  default     = "latest"
```

**Pull request merged**